src/buffer_object.rs

```diff
@@ -678,18 +634,10 @@ impl error::Error for InvalidFdError {}
 /// Thrown when an error occurs during getting a bo fd
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum FdError {
```

Comment:
Hmm, this make the error type quite unnecessary. Can we just re-use the `InvalidFdError` and drop this?